### ISSUE TITLE
Custom loss got nan after net.hybridize()

### ISSUE THREAD

**Issue by wanxinzzz** [dmlc/gluon-cv#1330]
Hello, I'm trying to implement giou loss base on gluoncv yolov3. But when i use `net.hybridize()`, the num of loss was multiplied several times and then it becomes `nan` after hundreds of iterations
```
# with net.hybridize()
[Epoch 0][Batch 199], LR: 1.00E-05, Speed: 24.338 samples/sec, ObjLoss=11.041, IouLoss=9.980, ClassLoss=3.658
[Epoch 0][Batch 999], LR: 1.00E-05, Speed: 18.101 samples/sec, ObjLoss=nan, IouLoss=nan, ClassLoss=nan
# without
[Epoch 0][Batch 199], LR: 1.00E-05, Speed: 25.527 samples/sec, ObjLoss=8.534, IouLoss=1.984, ClassLoss=2.495
```
Then i just replace F.elemwise_sub to F.broadcast_sub
```
# giou = F.elemwise_sub(iou, F.elemwise_div(F.elemwise_sub(convex, union), convex))
giou = F.broadcast_sub(iou, F.broadcast_div(F.broadcast_sub(convex, union), convex))
loss = 1. - giou
```
and run my code, everything seems fine, i got this tips
```
../src/imperative/cached_op.cc:192: Disabling fusion due to altered topological order of inputs.
```
What's the difference between the two op? And what's the mean of `Disabling fusion due to altered topological order of inputs` that make my code work
